src/main/java/ch/njol/skript/effects/EffContinue.java

```diff
@@ -65,35 +63,56 @@ public class EffContinue extends Effect {
 		);
 	}
 
-	@SuppressWarnings("NotNullFieldNotInitialized")
-	private LoopSection loop;
-	@SuppressWarnings("NotNullFieldNotInitialized")
-	private List<LoopSection> innerLoops;
+	private @UnknownNullability LoopSection loop;
+	private @UnknownNullability List<LoopSection> innerLoops;
+	private int breakLevels;
 
 	@Override
 	@SuppressWarnings("unchecked")
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		List<LoopSection> currentLoops = getParser().getCurrentSections(LoopSection.class);
+		List<TriggerSection> sections = getParser().getCurrentSections();
+		innerLoops = new ArrayList<>();
+		int loopLevels = 0;
+		LoopSection lastLoop = null;
 
-		int size = currentLoops.size();
-		if (size == 0) {
-			Skript.error("The 'continue' effect may only be used in loops");
+		int level = matchedPattern == 0 ? -1 : ((Literal<Integer>) exprs[0]).getSingle();
+		if (matchedPattern != 0 && level < 1) {
+			Skript.error("Can't continue the " + StringUtils.fancyOrderNumber(level) + " loop");
 			return false;
 		}
 
-		int level = matchedPattern == 0 ? size : ((Literal<Integer>) exprs[0]).getSingle();
-		if (level < 1) {
-			Skript.error("Can't continue the " + StringUtils.fancyOrderNumber(level) + " loop");
+		for (TriggerSection section : sections) {
```

Comment:
I think it would be better to iterate over this list *backwards* both for readability and performance. Also, is this correct given:
```
loop {_a::*}:
    loop {_b::*}:
        continue
```
It seems like break levels would end up as `2` (given the check at the end of init too), but shouldn't it be `1`?